Ds 504

Fennersteel opened the pull request.

Fennersteel committed:
Began conversion of script from bash to python

Fennersteel committed:
Added env info

Fennersteel committed:
WIP

Fennersteel committed:
Updated test_irods_unixfilesystem_resource to verify that resource and all attributes were properly instantiated

Fennersteel committed:
wip

tedgin committed:
HOTFIX: Allowing max irods direct conns to be configurable

tedgin committed:
Merge branch 'prod' into main

tedgin committed:
HOTFIX: Allowing max irods direct conns to be configurable

tedgin committed:
HOTFIX: Corrected shell call in admin/tune_networking

Fennersteel committed:
wip

tedgin committed:
Merge branch 'main' into DS-504

Fennersteel committed:
Finished conversion of irods_unix_filesystem to python 3.

Fennersteel committed:
Merge branch 'DS-504' of https://github.com/cyverse/ds-playbooks into DS-504

Fennersteel committed:
Fixed pylint conflicts

Commit:
DS-466: this removes firewall.yml, its reference in main.yml, and the block_new_connections.yml which relied on an earlier version of it

Commit:
DS-466: removed the remaining firewall stuff

Commit:
DS-466: documented playbooks in README

Commit:
Merge branch 'DS-466' into main

tedgin commented on irods/library/tests/test_irods_unixfilesystem_resource.yml:
```diff
@@ -1,7 +1,84 @@
+
+    - name: check resource name
+      failed_when: response is failed
+      shell:
```
To ensure all lines of the inline shell script are read, `shell` needs to be called with a `|` on its line to let it know that the subsequent indented lines are to be treated as individual lines and not a single line that has be wrapped.  

```yaml
shell: |
  iquest ...
  iquest ...
```

`shell` should always be called that way.

tedgin commented on irods/library/tests/test_irods_unixfilesystem_resource.yml:
```diff
@@ -1,7 +1,84 @@
+    - name: check resource name
+      failed_when: response is failed
+      shell:
+        iquest "SELECT RESC_NAME WHERE RESC_NAME = '{{ TEST_RSC_NAME }}'"
```
iquest doesn't fail if it doesn't find any results. It returns the response `CAT_NO_ROWS_FOUND: Nothing was found matching your query`.  To use iquest to verify the existence of something, you need to compare its response to the above one.

tedgin commented on irods/library/tests/test_irods_unixfilesystem_resource.yml:
```diff
@@ -1,7 +1,84 @@
+      register: response
+
+    - name: check resource name
+      failed_when: response is failed
```
The default is for a task to fail when the response has the  `failed` field set to `true`.  You don't need to override the default in this case.

tedgin commented on irods/library/tests/test_irods_unixfilesystem_resource.yml:
```diff
@@ -1,7 +1,84 @@
+        set -o errexit
+        iinit "$IRODS_PASSWORD"
+
   tasks:
```
This section would be more clear if you organized it in the following way.
1. run module without optional vars
1. verify that the storage resource was created correctly
1.  run module with all optional vars
1. verify that second storage resource was created correctly

tedgin commented on irods/library/tests/test_irods_unixfilesystem_resource.yml:
```diff
@@ -1,7 +1,84 @@
+    - name: check resource context
+      failed_when: response is failed
+      shell:
+        iquest "SELECT RESC_CONTEXT WHERE RESC_NAME = '{{ TEST_RSC_NAME }}' AND RESC_CONTEXT = '{{ TEST_CONTEXT }}'"
```
You didn't set a context for `REST_RESC_NAME`, so this test should have failed.

tedgin commented on irods/library/tests/test_irods_unixfilesystem_resource.yml:
```diff
@@ -1,7 +1,84 @@
+    TEST_VAULT_PATH2: /irods/test_vault2
+    TEST_CONTEXT: na
+    TEST_STATUS: up
+    TEST_INIT_FREESPACE: false
```
It doesn't make sense to test the default value.  Please set to `true`.

tedgin commented on irods/library/tests/test_irods_unixfilesystem_resource.yml:
```diff
@@ -1,7 +1,84 @@
+        zone: testing
+      register: response
+
+    - name: run script with optional vars
```
You need to verify that free space was initialized.

tedgin commented on irods/library/irods_unixfilesystem_resource:
```diff
@@ -1,195 +1,185 @@
-
-set -e
-main "$@" 2>&1
+#!/usr/bin/env python3
```
When the module is run by the test, it doesn't appear to set the `changed` field to `true`.

Fennersteel committed:
wip

Fennersteel committed:
Updated irods_unixfilesystem_resource and test play

- irods_unixfilesystem_resource
	- Updated "free_space" variable to be an integer of how much space the user would like allocated to the resource"

- test irods_unixfilesystem_resource.yml
	- Updated tests to verify resources and all attributes were properly instantiated

Commit:
Merge branch 'main' into DS-504

Fennersteel committed:
Rebase error fixes
